File: app/controllers/ajax/orders_controller.rb
# frozen_string_literal: true

class Ajax::OrdersController < ApplicationController
  def index
    @patient = Patient.find_by(id: params[:patient_id])

    @orders =
      if params[:canceled].to_i.zero?
        @patient.orders_only_active.page(params[:page])
      else
        @patient.orders.page(params[:page])
      end
    render layout: false
  end

  def edit
    @order = Order.find_by(id: params[:id])
    @originator = Employee.originator_of(@order)
    render layout: false
  end
end

File: app/bundles/javascripts/controllers/ajax/orders_controller.js
import { Controller } from 'stimulus';
import axios from 'axios';

export default class extends Controller {
  index(event) {
    const canceled = event.target.value;
    const patient_id = this.data.get('patientId');
    const page = this.data.get('page');

    axios.get('/ajax/orders', {
        params: { patient_id, canceled, page }
      })
      .then(response => response.data)
      .then(html => document.querySelector('#orders-listup').innerHTML = html);
  }

  edit(event) {
    event.preventDefault();
    const id = this.data.get('id');

    axios.get(`/ajax/orders/${id}/edit`)
      .then(response => response.data)
      .then(html => document.querySelector('#order-edit-modal').innerHTML = html);
  }
}
